File: LogAndroid/TrustarApp/src/main/kotlin/jp/co/toukei/log/trustar/repo/NoticeRepo.kt
package jp.co.toukei.log.trustar.repo

import io.reactivex.rxjava3.core.Flowable
import jp.co.toukei.log.trustar.db.user.db.UserDB
import jp.co.toukei.log.trustar.db.user.entity.Notice

const val Important_Notice = 1
const val Normal_Notice = 2

class NoticeRepo(private val userDB: UserDB) {

    fun countUnreadImportant(): Flowable<Int> {
        return userDB.noticeDao().countUnreadByRank(Important_Notice)
    }

    fun countUnreadNormal(): Flowable<Int> {
        return userDB.noticeDao().countUnreadByRank(Normal_Notice)
    }

    /**
     * @param rank 1：重要、2：通常
     */
    fun unreadBy(rank: Int): Flowable<List<Notice>> {
        return userDB.noticeDao().unreadByRank(rank)
    }

    fun allByRank(rank: Int): Flowable<List<Notice>> {
        return userDB.noticeDao().selectByRank(rank)
    }

    fun markRead(rank: Int) {
        userDB.noticeDao().apply {
            markAllReadByRank(rank, System.currentTimeMillis())
        }
    }
}

File: LogiOS/TrustarApp/Repo/NoticeRepo.swift
//
//  NoticeRepo.swift
//  TrustarApp
//
//  Created by CuongNguyen on 03/03/2022.
//  Copyright © 2022 DionSoftware. All rights reserved.
//

import Foundation

class NoticeRepo {
    private var userDb: UserDb
    init(_ userDb: UserDb) {
        self.userDb = userDb
    }
    
    func countUnreadImportant() -> Int {
        return userDb.noticeDao!.countUnreadByRank(rank: 1)
    }

    func unreadImportant() -> [Notice] {
        return unreadBy(rank: 1)
    }

    func unreadNormal() -> [Notice] {
        return unreadBy(rank: 2)
    }

    /**
     * @param rank 1：重要、2：通常
     */
    func unreadBy(rank: Int) -> [Notice] {
        return userDb.noticeDao?.unreadByRank(rank: rank) ?? []
    }

    func allByRank(rank: Int) -> [Notice] {
        return userDb.noticeDao?.selectByRank(rank: rank) ?? []
    }

    func markRead(rank: Int) {
        userDb.noticeDao?.markAllReadByRank(rank: rank)
    }
}
